### PR TITLE
I3 custom errors

### LINKED ISSUE #3: Custom errors
Main task:
- [x] implement custom errors (introduced with  Solidity v0.8.4)
- [x] refactor the video agreement struct so it costs less 
Side tasks:
- [x] reduce the number of modifiers (when the modifier is used only once, use a simple check inside the function)
- [x] rework withdraw logic
- [x] use private key of an account instead of mnemonics of a wallet to prevent security issues

### PR DESCRIPTION
**Pull request by schronck**
Closes #3

### CODE REVIEW

File: contracts/Monetizer.sol [hunk 1]
```diff
@@ -7,114 +7,79 @@ import "witnet-ethereum-bridge/contracts/UsingWitnet.sol";
...
   /// @param _lockTime        the time to lock
   /// @param _targetViewCount the viewcount that is required for the withdrawal
   // prettier-ignore
   function deposit(
-    string calldata _id,
-    address payable _beneficiary,
-    uint256         _lockTime,
-    uint256         _targetViewCount
-  ) external payable empty(_id) {
+    bytes11 _id,
```
Comment (TomiOhl) [inline contracts/Monetizer.sol:57]: Little formatting mistake here

**Review by TomiOhl — APPROVED** [review]
Apart from that little formatting stuff lgtm

**Review by Rick-xyz — APPROVED** [review]
lgtm

File: contracts/Monetizer.sol (hunk 1 above)
Comment (schronck) [reply contracts/Monetizer.sol:57]: done

**Pull request merged**